CHANGELOG.md

```diff
@@ -77,10 +77,19 @@ Adding a new version? You'll need three changes:
   [#3832](https://github.com/Kong/kubernetes-ingress-controller/pull/3832)
 - Added license agent for Konnect-managed instances.
   [#3883](https://github.com/Kong/kubernetes-ingress-controller/pull/3883)
-- `Service`, `Route` and `Consumer` Kong entities now get assigned deterministic
+- `Service`, `Route`, and `Consumer` Kong entities now get assigned deterministic
   IDs based on their unique properties (name, username, etc.) instead of random
   UUIDs.
+  After the upgrade to 2.10.0, the controller will re-create all the existing
+  entities (Services, Routes, and Consumers) with the new IDs. That can
+  potentially lead to temporary downtime between the deletion of the old entities
+  and the creation of the new ones.
+  Users might want to disable the feature if their existing DB-backed setup
+  consists of a huge amount of entities for which the recreation can take
+  significant time. It is possible to opt out of this behavior by setting
+  `FillIDs` feature gate to `false`.
```

Comment:
Given the disruption this flag can cause I'm on the fence with this one.

We're bumping a minor version to 2.10 where users with big configurations can experience severe consequence during the upgrade. I do recognize that that the upgrade will be non breaking in a sense that eventually we'll recover but I do hope that we're planning to make sure that users to take this changelog notes seriously and do read them.

Having said that: I definitely do not intend to block this change on these grounds but I'd suggest to think how big of a surprise this might be for some users.

cc @mheap as he might know users habits around upgrades the most.

Reply:
I'd prefer to default to false. Possible downtime is not something we want, especially as we have KIC 3.0 coming which is a better time to default true.